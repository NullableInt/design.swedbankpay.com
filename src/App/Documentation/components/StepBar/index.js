import React from "react";
import PrismCode from "react-prism";

import { ComponentPreview, DocToc, ExperimentalComponentAlert, Property } from "#";
import StepBarComponent from "@/StepBar";

const BasicStepBar = () => {
    const items = [
        {
            title: "Step one",
            subtitle: "24.12.17 12:10",
            completed: true

        },
        {
            title: "Step two",
            active: true
        },
        {
            title: "Step three"
        },
        {
            title: "Step four"
        }
    ];

    return (
    <>
        <h2 id="basic-usage">Basic usage</h2>
        <p>Add class <Property value=".step-bar" /> to an <PrismCode className="language-html">{"<ol>"}</PrismCode> element.</p>
        <ComponentPreview language="html" showCasePanel codeFigure>
            <StepBarComponent items={items} />
        </ComponentPreview>
    </>
    );
};

const HorizontalStepBar = () => {
    const items = [
        {
            title: "Step one",
            completed: true
        },
        {
            title: "Step two",
            active: true
        },
        {
            title: "Step three"
        },
        {
            title: "Step four"
        }
    ];

    return (
    <>
        <h2 id="horizontal-step-bar">Horizontal step bar</h2>
        <p>In addition to class <Property value=".step-bar" />, add the extension class <Property value=".step-bar--horizontal" /> to an <PrismCode className="language-html">{"<ol>"}</PrismCode> element.</p>
        <ComponentPreview language="html" showCasePanel codeFigure>
            <StepBarComponent items={items} horizontal />
        </ComponentPreview>
    </>
    );
};

const StepBarText = () => (
    <div className="col-lg-10 doc-body">
        <ExperimentalComponentAlert />
        <p className="lead">Step Bar.</p>
        <BasicStepBar />
        <HorizontalStepBar />
    </div>
);

const StepBar = () => (
    <div className="doc-container">
        <div className="row">
            <StepBarText />
            <DocToc component={StepBarText} />
        </div>
    </div>
);

export default StepBar;

/* for testing */
export { BasicStepBar, HorizontalStepBar, StepBarText };
