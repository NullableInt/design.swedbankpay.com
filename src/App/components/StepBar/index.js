import React from "react";
import PropTypes from "prop-types";
import classnames from "classnames";

const DefaultItems = () => (
    <div>
        <li className="step-bar-item-complete">
            <div className="material-icons step-bar-icon" >check</div>
            Step one
        </li>
        <li className="step-bar-item-active">
            Step two
        </li>
        <li>
            Step three
        </li>
        <li>
            Step four
        </li>
    </div>
);

const StepBar = ({ items, horizontal }) => {
    const stepBarClasses = classnames(
        "step-bar",
        horizontal ? "step-bar--horizontal" : null
    );

    const _renderItems = items => items.map(({ title, subtitle, completed, active }, i) => (
        <li key={i} className={`${completed ? "step-bar-item-complete" : ""}${active ? " step-bar-item-active" : ""}`}>
            {completed ? <div className="material-icons step-bar-icon" >check</div> : null}
            {title}
            {subtitle ? <div className="step-bar-item-sub-title">{subtitle}</div> : null}
        </li>
    ));

    return (<ol className={stepBarClasses}>
        {items ? _renderItems(items) : <DefaultItems />}
    </ol>
    );
};

StepBar.propTypes = {
    items: PropTypes.array,
    horizontal: PropTypes.bool
};

export default StepBar;

// exporting for testing purposes
export { DefaultItems };
