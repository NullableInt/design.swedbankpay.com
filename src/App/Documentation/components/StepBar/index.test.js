import React from "react";
import { shallow } from "enzyme";
import StepBar, { BasicStepBar, StepBarText } from "./index";

describe("Documentation: StepBar", () => {
    it("is defined", () => {
        expect(StepBar).toBeDefined();
    });

    it("renders", () => {
        const wrapper = shallow(
            <StepBar />
        );

        expect(wrapper).toMatchSnapshot();
    });

    describe("Basic StepBar", () => {
        it("is defined", () => {
            expect(BasicStepBar).toBeDefined();
        });

        it("renders", () => {
            const wrapper = shallow(
                <BasicStepBar />
            );

            expect(wrapper).toMatchSnapshot();
        });
    });

    describe("StepBar Text", () => {
        it("is defined", () => {
            expect(StepBarText).toBeDefined();
        });

        it("renders", () => {
            const wrapper = shallow(
                <StepBarText />
            );

            expect(wrapper).toMatchSnapshot();
        });
    });
});
