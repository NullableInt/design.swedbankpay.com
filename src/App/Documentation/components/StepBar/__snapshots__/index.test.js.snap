// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Documentation: StepBar Basic StepBar renders 1`] = `
<Fragment>
  <h2
    id="basic-usage"
  >
    Basic usage
  </h2>
  <p>
    Add class 
    <Property
      value=".step-bar"
    />
     to an 
    <PrismCode
      className="language-html"
      component="code"
    >
      &lt;ol&gt;
    </PrismCode>
     element.
  </p>
  <ComponentPreview
    codeFigure={true}
    language="html"
    showCasePanel={true}
  >
    <StepBar
      items={
        Array [
          Object {
            "completed": true,
            "subtitle": "24.12.17 12:10",
            "title": "Step one",
          },
          Object {
            "active": true,
            "title": "Step two",
          },
          Object {
            "title": "Step three",
          },
          Object {
            "title": "Step four",
          },
        ]
      }
    />
  </ComponentPreview>
</Fragment>
`;

exports[`Documentation: StepBar StepBar Text renders 1`] = `
<div
  className="col-12 col-lg-10 doc-body"
>
  <ExperimentalComponentAlert />
  <p
    className="lead"
  >
    Step Bar.
  </p>
  <BasicStepBar />
  <HorizontalStepBar />
</div>
`;

exports[`Documentation: StepBar renders 1`] = `
<div
  className="doc-container"
>
  <div
    className="row"
  >
    <StepBarText />
    <DocToc
      component={[Function]}
    />
  </div>
</div>
`;
